Adding a note on dvc plots diff --targets with revisions usage (#1549)
iterative/dvc.org: content/docs/command-reference/plots/diff.md

- `--targets <path>` - specific metric files to visualize. These must be listed
  in a [`dvc.yaml`](/doc/user-guide/dvc-files-and-directories#dvcyaml-file) file

>   (see the `--plots` option of `dvc run`). When you are specifying multiple
>   arguments for `--targets` and `revisions`, use `--` after all the arguments
>   for `--targets`:

Reviewer: ```suggestion
  (see the `--plots` option of `dvc run`). When specifying multiple targets and
  `revisions`, you may use `--` after this option's arguments, e.g.:
```

Author: I'll incorporate these changes in a new PR

Reviewer: I see it's done in #1567, thanks.

  ```dvc
  $ dvc plots diff --targets t1.json t2.csv -- HEAD v1 v2
  ```

>   Without `--`. this statement will show an error as argument parser confuses
>   `revisions` as arguments for `--targets` option. Alternatively, you can also
>   run above statement as:

Reviewer: Just
```suggestion
Alternatively, you can also run above statement as:
```

Reviewer: I wonder is this should be the first recommendation though, not `--`. I also wonder which other commands can benefit from the `--` syntax.

Reviewer: WDYT @imhardikj ?

Author: I think order is correct. As mentioned in https://github.com/iterative/dvc.org/pull/1544#discussion_r451395365 it's almost natural instinct to give `targets` first and then `revisions`.

Author: `--` will be used only when a option accepts multiple arguments and at least one positional argument. This can be used in `metrics diff`. But right now `metrics diff` doesn't have any example which shows usage of `--targets` option.

Reviewer: But the same note applies to metrics diff? If so please copy it to that reference as well in your new PR, or a subsequent one 🙂 Adding an targets example (similar to the one here) would be a nice addition too but not necessary.

Compare two specific versions (commit hashes, tags, or branches):

```dvc
$ dvc plots diff --targets logs.csv -- HEAD 0135527
file:///Users/usr/src/plots/logs.csv.html
```

![](/img/plots_diff.svg)

> Please note that when you're specifying `--targets` options with multiple
> `revisions`:
>
> ```dvc
> $ dvc plots diff --targets t1.json t2.csv HEAD v1 v2
> ```
>
> This statement will show an error because argument parser confuses `revisions`
> as arguments for `--targets` option. To avoid this error use `--` after all

> > the arguments for `--targets`:

Reviewer: Let's move this note up to the description as a regular paragraph, and just make a quick reminder in this note (block quote) here in the example.

Reviewer: Actually, per https://github.com/iterative/dvc.org/pull/1544#discussion_r452406892 please move up into the `--targets` option bullet text instead. Feel free to use several paragraphs and even code blocks or block quote notes in there, but do try to summarize all this a little in both places. Thanks

Reviewer 2: please, let's try to keep it really simple. The problem itself is very simple and clear and one sentence should be enough in the description or option description to my mind.

Reviewer: Yeah, by "summarize all this a little" I meant summarize it a lot too 🙂

Reviewer: Continued in https://github.com/iterative/dvc.org/pull/1549#pullrequestreview-446819554

>
> ```dvc
> $ dvc plots diff --targets t1.json t2.csv -- HEAD v1 v2
> ```
>
> Alternatively, you can also run above statement as:
>
> ```dvc
> $ dvc plots diff HEAD v1 v2 --targets t1.json t2.csv

> > ```

Reviewer: This block quote is way too long now :P

> > ```

Author: @jorgeorpinel is this part needed in `--target` option. I think rest of the part can be moved up as it and here instead we can use example in line 108 itself -> `$ dvc plots diff --targets logs.csv -- HEAD 0135527`. Along with one line note reminding to use `--`. 
Thoughts?

Reviewer: I'm not sure, was this part of the example before? What about just use this one for the example (no --) and remove the note? Just try to summarize everything as much as possible, make your change, and we'll review that again. Thanks
